public/app/plugins/datasource/loki/LogContextProvider.ts

```diff
@@ -118,7 +91,7 @@ export class LogContextProvider {
     this.onContextClose =
       this.onContextClose ??
       (() => {
-        this.prepareContextExpr = this.prepareContextExprWithoutParsedLabels;
+        delete this.contextFilters[row.uid];
```

Comment:
I kind of like that we are saving filters per row. Do you think it would be fine to not delete them here and just keep them?

Reply:
We could do that. Are you also thinking showing/re-using selected filters if users closes and then re-opens the same row?

Reply:
Oh, I see. I thought we'll get that logic already if we just don't delete the entry here, but you are right, that's not super easy.

I already heard early feedback that it's a painpoint for users to always remove/add the same labels, so something which "remembers" what labels were included would be great - but it's also something that can be a future iteration, I think.